rfcs/2020-09-03-3226-api-topology.md

```diff
@@ -0,0 +1,141 @@
+# RFC #3226 - 2020-09-03 - API Topology
+
+This RFC discusses configuration, implementation and potential schema for the API to co-exist with, and observe, existing Vector topology.
+
+## Scope
+
+* Implementation of the API server alongside existing topology.
+* Configuration in `vector.toml` to set-up / enable the API.
+* GraphQL types to observe existing topology (pending #3648)
+
+## Motivation
+
+I implemented the first draft of a GraphQL API server in #3514, and made changes to the toplogy mod to accommodate it. @lukesteensen [commented](https://github.com/timberio/vector/pull/3514#discussion_r479517596) with a preference to keep the API out of the existing topology mod.
+
+This raises several design questions:
+
+* Where should the API live?
+* If the API is outside of topology, what are our expectations with users making API configuration changes in `vector.toml` and handling SIGHUP/reloading? (The initial draft in #3514 )
```

Comment:
I propose that we ignore config reloading. There are already config values we don't support changing with reloading and I don't think it's terribly important that users be able to toggle this dynamically.

This also feels like something that's low risk to add in the future. If we decide it's worthwhile to hook into reloading, we can do it later and it shouldn't be any more difficult than doing it now.